### ISSUE TITLE
Character Editor

### ISSUE THREAD

**Issue by goblinJoel** [rpgtoolkit/editor#21]
The character editor has been started on the dev-character-editor branch. It is partially complete.

The stats/portrait panel was working according to my commit log from early May.

This branch includes some Gui methods for image/icon loading and creation. The loading could probably be updated to use the assets pipeline.

**Comment by goblinJoel** [comment]
Merged develop in to try and keep things from getting too crazy divergent.

I've got the special moves panel going and have started working on the equipment panel. After that, it'll be levels (which is pretty straightforward) and graphics.

@swordmaster2k I think the graphics tab is going to be a bit beyond me. Since you did the board editor, are you interested in working on that part of the character editor?

**Comment by swordmaster2k** [comment]
Yeah, I'll take a look into the graphics side of things.

**Comment by goblinJoel** [comment]
Great! Feel free to check out the character editor branch for this. I just made sure all my current changes were pushed.

**Comment by goblinJoel** [comment]
@swordmaster2k just a heads up as I look through the updates you've made to the character editor, you may want to look at the character editor branch on the core project, too. Looks like you're undoing certain things that probably looked broken if you were using the develop branch on core. (For example, Player.getHasSpecialMoves() converts the byte to a boolean on the character editor branch, while you changed the editor's use of it back to expecting a byte.)

**Comment by swordmaster2k** [comment]
Looking back over the History of the Player class I can only ever see it
returning a byte:

```
public byte getHasSpecialMoves() {
    return hasSpecialMoves;
}
```

And the Character editor code read:

if (this.player.getHasSpecialMoves() == false) {
      this.usesSpecials.doClick();
    }

Do you have some uncommited code changes? I checked out both branches and I
remember 1 or 2 issues like this that I had to fix before merging them back
into the develop stream.

**Comment by goblinJoel** [comment]
Not uncommitted, but never merged into develop, probably.

Branch: https://github.com/rpgtoolkit/core/tree/dev-character-editor
Source: https://github.com/rpgtoolkit/core/blob/dev-character-editor/src/main/java/net/rpgtoolkit/common/assets/Player.java

In a few places (don't recall them all), I'd decided that it was silly to be playing with bytes when those bytes were really booleans. So I changed the getters and setters to convert, thinking maybe we can change the actual fields to booleans later and let legacy i/o convert to bytes if it really wants to.

**Comment by swordmaster2k** [comment]
Yeah I forgot to merge those changes in. I was working on the Project
editor prior to the Character editor and I probably mixed up what branches
I had merged into where. Anyway I've switched over to both branches and
merged develop's changes in.

Thanks for pointing that out.

**Comment by goblinJoel** [comment]
Awesome. Hopefully the merge was painless. Thanks for sprucing up the
character editor!

**Comment by swordmaster2k** [comment]
Yeah the merge was pretty painless thankfully. I've made a couple of changes with the character editor. I replaced the JLabel for showing animations with a dedicated custom panel that can be reused in the Enemy editor and Animation editor etc. I also replaced the ListView with a JTable and a AnimationTableModel.

Also working on implementing a change listener/model, basically enabling the animations table to detect updates via the UI in accordance with MVC. It would also enable the character editor to detect changes for saving etc.
